Source/JavaScriptCore/builtins/JSIteratorPrototype.js

```diff
@@ -138,15 +150,21 @@ function drop(limit)
     if (!@isObject(this))
         @throwTypeError("Iterator.prototype.drop requires that |this| be an Object.");
 
-    var numLimit = @toNumber(limit);
-    if (numLimit !== numLimit)
+    var iterated = this;
```

Comment:
Let's just use `this` directly in this function instead of `iterated` when possible.